src/DynamoCoreWpf/Views/PackageManager/PackageManagerView.xaml.cs

```diff
@@ -117,12 +124,55 @@ private void WindowClosed(object sender, EventArgs e)
         {
             this.packageManagerPublish.Dispose();
             this.PackageManagerViewModel.PackageSearchViewModel.RequestShowFileDialog -= OnRequestShowFileDialog;
+            this.PackageManagerViewModel.PackageSearchViewModel.Close();
         }
 
         private void SearchForPackagesButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Search for Packages tab
+            Navigate(Dynamo.Wpf.Properties.Resources.PackageManagerSearchTab);
+        }
+
+        private void PublishPackageButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Publish a Package tab
+            Navigate(Dynamo.Wpf.Properties.Resources.PackageManagerPublishTab);
+        }
+
+        /// <summary>
+        ///  Navigatest to the selected tab
+        /// </summary>
+        /// <param name="tabName">Tab name to navigate to</param>
+        internal void Navigate(string tabName)
         {
             var tabControl = this.projectManagerTabControl;
-            tabControl.SelectedIndex = 0;
+
+            var preferencesTab = (from TabItem tabItem in tabControl.Items
+                                  where tabItem.Header.ToString().Equals(tabName)
+                                  select tabItem).FirstOrDefault();
+            if (preferencesTab == null) return;
+            tabControl.SelectedItem = preferencesTab;
+        }
+
+        private void NavigateToPreferencesPanel()
+        {
+            var tabName = Dynamo.Wpf.Properties.Resources.PreferencesPackageManagerSettingsTab;
+            var expanderName = Dynamo.Wpf.Properties.Resources.PackagePathsExpanderName;
+            PreferencesPanelUtilities.OpenPreferencesPanel(dynamoView, WindowStartupLocation.CenterOwner, tabName, expanderName);
```

Comment:
Do we do this so it works in a localized environment? But I think it is the other way round right? I would fail that way, did I miss anything here

Reply:
I can see that you are following an existing example, let's leave it for now